controllers/horizon_controller.go

```diff
@@ -625,22 +614,30 @@ func (r *HorizonReconciler) renderMemcached(instance *horizonv1alpha1.Horizon) *
 	}
 }
 
-func (r *HorizonReconciler) getMemcachedSvc(ctx context.Context, memcachedName string, h *helper.Helper, namespace string) (svc string, err error) {
-	labelSelector := k8s_labels.Set{"app": "memcached"}
-	svcList, err := oko_svc.GetServicesListWithLabel(ctx, h, namespace, labelSelector)
+func validateHorizonSecret(secret *corev1.Secret) bool {
+	return len(secret.Data["horizon-secret"]) != 0
+}
 
-	if err != nil {
-		r.Log.Info("Error getting service list: %s", err)
-		return "", err
+func getMemcachedPods(ctx context.Context, h *helper.Helper, instance *horizonv1alpha1.Horizon, memcachedName string) ([]string, error) {
```

Comment:
getMemcachedHosts would be better hear because it returns list of hostnames

Reply:
`getMemcachedPodFQDNs`?